adr/001-component-library.md

```diff
@@ -0,0 +1,65 @@
+# Introduce component library
+
+- Status: Accepted
+- Deciders: Team
+- Date: 17.10.2024
+
+## Result
+
+A1: Component library is introduced, firstly as just a folder in our current setup, adding yarn workspaces or similar requires more research and testing
+
+## Problem context
+
+Today our UI components are tightly coupled to the app in which they are rendered.
+
+This leads to several issues:
+
+- Makes it hard to do component testing outside a fully rendered app.
+  - See tests in `test/e2e/integration` for examples of how we have most of our tests in integration tests because components need to be running inside an app to work.
+  - See `src/test/renderWithProviders.tsx` for examples of how much scaffolding we need to do to run component tests currently. This also leads to very slow component tests.
+- Makes refactoring the app framework a lot harder.
+  - If we search for `useNodeItem` in src/layout we get over 100 hits. If we make changes or remove hooks like this, every single UI component must be updated.
+- Leads to unclear interfaces between UI components and the app framework.
+  - See: `src/layout/Button/ButtonComponent.tsx`. This component references `node.parent` which confuses the role and responsibility of the button component.
+- Makes developing UI components complex without deep understanding of the application.
+- Enables sharing of pure components (docs, Studio, storybook)
+
+## Decision drivers
+
+A list of decision drivers. These are points which can differ in importance. If a point is "nice to have" rather than
+"need to have", then prefix the description.
+
+- B1: UI components should only receive data to display, and notify the app when data is changed.
+- B2: UI components should live in a separate folder from the app itself, and have no dependencies to the app.
+- B3: UI components should live in a lib separately to the src folder to have stricter control of dependencies.
```

Comment:
Ah, maybe it's my misunderstanding of ADRs. This is just documenting everything you talked about, even though the final solution fell short of supporting B3? Still, I didn't immediately see the connection to the problem context.

Reply:
So, the ADR conculdes that we should not introduce yarn workspaces or NX at this point, that would be separate ADR if we decied to investigate further. So that would as you say be pulled into a separate ADR.

But youre right, it could make sense that this ADR asked "should we introduce a component library?". However this was really already agreed upon in frontend architecture forum, so this ADR basically formulates _how_ what was already agreed upon should be implemented.